.github/workflows/pipeline-definition.yaml

```diff
@@ -94,53 +111,92 @@ jobs:
           echo "VERSION: ${{steps.set-outputs.outputs.version}}"
           echo "GH_USERNAME: ${{steps.set-outputs.outputs.gh_username}}"
 
-  # Check go unit tests
-  test_go:
-    timeout-minutes: 6
+  # Run all the linters.
+  linters-run:
+    name: Run all linters
+    if: inputs.run_linters
+    needs:
+      - setup
+    uses: ./.github/workflows/linters.yaml
+    secrets: inherit
+    with:
+      golang_version: ${{ needs.setup.outputs.golang_version }}
+
+  # Check the result of all linters and fails if any linter failed.
+  linters-result:
+    name: Check linters result
+    if: inputs.run_linters && always()
+    needs: linters-run
+    runs-on: ubuntu-latest
+    steps:
+      # Fail the job if any linter did not succeed.
+      - name: Check all linters result
+        if: needs.linters-run.result != 'success'
+        run: exit 1
+
+  # Check go unit tests.
+  test-with-go:
+    name: Check Go test suite
     needs:
       - setup
     runs-on: ubuntu-latest
     steps:
+      # Checkout the repository.
       - uses: actions/checkout@v5
         name: Checkout
+      # Read tool versions for environment setup.
```

Comment:
question:
why are you parsing the tool-version if you are not using afterwards? (you are getting the version from previous steps)

Reply:
ditto, I did want to add too many overlapping changes with https://github.com/trento-project/mcp-server/pull/25